zebra-test/src/command.rs

```diff
@@ -17,23 +17,29 @@ pub fn test_cmd(command_path: &str, tempdir: &PathBuf) -> Result<Command> {
     Ok(cmd)
 }
 
-/// Create a temp directory and optional config file
-pub fn tempdir(create_config: bool) -> Result<(PathBuf, impl Drop)> {
+/// Create a temp directory and optional default config file with optional additional options
+pub fn tempdir(
+    create_config: bool,
+    additional_config: Option<String>,
+) -> Result<(PathBuf, impl Drop)> {
     let dir = TempDir::new("zebrad_tests")?;
 
     if create_config {
         let cache_dir = dir.path().join("state");
         fs::create_dir(&cache_dir)?;
-        fs::File::create(dir.path().join("zebrad.toml"))?.write_all(
-            format!(
-                "[state]\ncache_dir = '{}'\nmemory_cache_bytes = 256000000",
-                cache_dir
-                    .into_os_string()
-                    .into_string()
-                    .map_err(|_| eyre!("tmp dir path cannot be encoded as UTF8"))?
-            )
-            .as_bytes(),
-        )?;
+
+        let mut content: String = format!(
+            "[state]\ncache_dir = '{}'\nmemory_cache_bytes = 256000000\n",
+            cache_dir
+                .into_os_string()
+                .into_string()
+                .map_err(|_| eyre!("tmp dir path cannot be encoded as UTF8"))?
+        );
+        if additional_config.is_some() {
+            content = format!("{}{}", content, additional_config.unwrap());
+        }
```

Comment:
Not in love with the string manipulation, and I expect this will only get more complicated. I think it would really be better to just have a `.with_config(Config { ... })` function that lets us manipulate the config more naturally rather than having to build up toml by hand.

Reply:
I agree with this, you even saw in the past this was going to be a problem so i think we should do it but maybe better in a different PR. Maybe i should open an issue for it ?

Reply:
If we add the listen_addr to all configs, we can remove this code, which would also resolve the issue:
https://github.com/ZcashFoundation/zebra/pull/920/files#r473304292

Reply:
If we want to create a config by serialising a config object, let's do that in a separate issue and PR.

The serialisation won't work exactly the same way - we'll end up with a larger config containing all the defaults, rather than the limited config we're using right now. But the test behaviour should be the same, because the extra fields are defaults.

Reply:
I opened #929 to follow this up.